### PR TITLE
Bleeding messages now vary based on body part and intensity, changes message shown when applying pressure

### PR DESCRIPTION
**Pull request by Ilysen**
#### Summary
Interface "Messages displayed from bleeding will display the name of the bleeding part, and vary based on intensity"

#### Purpose of change

Right now, the only message displayed while bleeding is `You lose some blood.`, regardless of how bad the bleeding is or where it's located. I personally felt like it was a bit lackluster for someone who was presumably spurting gouts of the stuff from a wound.

#### Describe the solution

Bleeding messages now describe the body part and severity. A minor wound on your left arm, for instance, might display `Blood drips from your left arm.`, while a very bad one on your head might say `Blood pours from your head!` It does this using a const map of integers (referencing intensity) and strings, and iterates it in reverse to find the first entry where the integer is at or above the intensity of the effect.

NPCs still display the original message (`<npcname> loses some blood.`) because I wasn't sure how to have NPC ones displayed in a clean way, but I'm totally open to suggestions. Additionally, the generic message remains as a fallback in case there are no body parts assigned to the effect, as otherwise there's a hard crash.

While I was around, I also touched up some other related things:
* Bleeding now has a remove message: `The bleeding stops!`
* Fixed some grammar issues with a few descriptions in the bleeding and hypovolemia effects.
* Changed the message displayed when putting pressure on a wound to emphasize that it's something the player is doing over time that they are supposed to do repeatedly: `You attempt to put pressure on the bleeding wound!` -> `You put pressure on the bleeding wound…`
* If a character has acid blood, the message specifically references acid instead of regular blood, ala `Acid leaks from your torso.`

#### Describe alternatives you've considered

I was originally going to rename `Heavy Arterial Bleeding` to `Extreme Bleeding`, but I decided against it at the eleventh hour and removed that change from the commit.

#### Testing

Spawned in a giant dragonfly and let it shred me a bunch. Viewed the messages shown by the effects to verify that they gradually changed as intensity dropped. Used debug to apply the bleed effect with varying durations and saw that the fallback message was being used correctly.

As a known issue: putting pressure on a debug-applied bleeding wound appears to cause a hard crash. This is a bug that I was able to reproduce on vanilla, without the local changes. This PR doesn't cause this problem, but I figured it was a good opportunity to mention that it isn't something caused by it in case someone wants to test it locally and discovers the problem.

#### Additional context

cataclysm-tiles_vIl8EvVpyk

### CODE REVIEW

**Comment by NetSysFire** [comment]
Cool! I would potentially use the more ominous red for the heavy and very dangerous bleeding to tell the player it requires immediate attention.

**Comment by Ilysen** [comment]
That's actually pulled from the message log (`P`) and not the actual chat window itself, so the colors are a bit inaccurate. Is there a message span/color that displays dark red from the get-go, and not light red like the normal `bad` message type?

**Comment by Qrox** [comment]
I think you can just wrap the text within `<color_red>` and `</color>` tags.

File: src/player_hardcoded_effects.cpp [hunk 1]
```diff
@@ -284,8 +285,36 @@ static void eff_fun_bleed( Character &u, effect &it )
...
+            const std::string blood_str = !u.has_trait( trait_ACIDBLOOD ) ? "Blood" : "Acid";
+            // the numerical values here coincide with the intensity thresholds at which the name of the effect changes
+            // i.e. 0-5 intensity is displayed as "Minor Bleeding", 11-20 intensity is displayed as "Bad Bleeding", etc
+            const std::map<int, std::string> intensity_strings = {
+                { 0, "%1s drips from your %2s." },
+                { 6, "%1s leaks from your %2s. " },
+                { 11, "%1s flows from your %2s." },
+                { 21, "%1s pours from your %2s!" },
+                { 31, "%1s sprays from your %2s!" }
+            };
```
Comment (Qrox) [inline src/player_hardcoded_effects.cpp:297]:
```suggestion
            static const std::map<int, translation> intensity_strings = {
                { 0, to_translation( "%1s drips from your %2s." ) },
                { 6, to_translation( "%1s leaks from your %2s. " ) },
                { 11, to_translation( "%1s flows from your %2s." ) },
                { 21, to_translation( "%1s pours from your %2s!" ) },
                { 31, to_translation( "%1s sprays from your %2s!" ) }
            };
```

File: src/player_hardcoded_effects.cpp [hunk 2]
```diff
@@ -284,8 +285,36 @@ static void eff_fun_bleed( Character &u, effect &it )
             u.mod_pain( 1 );
         }
         if( one_in( 120 / intense ) ) {
+            const std::string blood_str = !u.has_trait( trait_ACIDBLOOD ) ? "Blood" : "Acid";
```
Comment (Qrox) [inline src/player_hardcoded_effects.cpp:288]:
```suggestion
            static const translation blood_str = !u.has_trait( trait_ACIDBLOOD ) ? to_translation( "bleed_message", "Blood" ) : to_translation( "bleed_message", "Acid" );
```

File: src/player_hardcoded_effects.cpp [hunk 3]
```diff
@@ -284,8 +285,36 @@ static void eff_fun_bleed( Character &u, effect &it )
...
+            }
             u.bleed();
-            u.add_msg_player_or_npc( m_bad, _( "You lose some blood." ),
+            bodypart_id bp = it.get_bp();
+            // piece together the final displayed message here instead of inline, for readability's sake
+            // format the chosen string with the relevant variables to make it human-readable, then translate everything we have so far
+            // we maintain a generic part-less fallback just in case the effect is added without a target body part, in order to avoid crashes
+            const std::string final_message = bp != bodypart_str_id::NULL_ID() ? string_format( _
+                                              ( suffer_string ),
+                                              _( blood_str ), body_part_name_accusative( bp ) ) : _( "You lose some blood." );
```
Comment (Qrox) [inline src/player_hardcoded_effects.cpp:314]:
```suggestion
            const std::string final_message = bp != bodypart_str_id::NULL_ID() ? string_format(
                                              suffer_string,
                                              blood_str, body_part_name_accusative( bp ) ) : _( "You lose some blood." );
```

**Review by Qrox — COMMENTED** [review]
Use `to_translation` to correctly extract the strings for translation.

**Comment by NetSysFire** [comment]
I would use "gushes" instead of "sprays" for the highest intensity of bleeding. Sprays sounds weird to me in that specific case.

**Comment by PatrikLundell** [comment]
If you're going for dark red messages, please check that they're actually legible, as dark red/blue on black can have a very poor contrast.

**Comment by NetSysFire** [comment]
The current color of the messages is `c_light_red` and I specifically mean `c_red`. These are default colors and I do not find them hard to read at all, so I think the contrast is going to be fine.

File: src/player_hardcoded_effects.cpp (hunk 1 above)
Comment (Ilysen) [reply src/player_hardcoded_effects.cpp:297]: Thanks; I know basically nothing about the translation systems other than "underscores and parentheses good". :P

**Comment by Ilysen** [comment]
I tried using color wrappers, but it seems like classifying the message as `bad` overrides it. In the interest of consistency, I'll stay with the default "bad" message class for now.

**Pull request merged**